crates/sui-graphql-rpc/src/types/type_filter.rs

```diff
@@ -132,6 +132,31 @@ impl FqNameFilter {
                 .filter(name.eq(n.clone())),
         }
     }
+
+    /// Try to create a filter whose results are the intersection of the results of the input
+    /// filters (`self` and `other`). This may not be possible if the resulting filter is
+    /// inconsistent (e.g. a filter that requires the module member's package to be at two different
+    /// addresses simultaneously), in which case `None` is returned.
+    pub(crate) fn intersect(self, other: Self) -> Option<Self> {
+        use FqNameFilter as F;
+        use ModuleFilter as M;
+
+        match (&self, &other) {
+            (F::ByModule(m), F::ByModule(n)) => m.clone().intersect(n.clone()).map(Self::ByModule),
+            (F::ByFqName(_, _, _), F::ByFqName(_, _, _)) => (self == other).then_some(self),
+
+            (F::ByFqName(p, _, _), F::ByModule(M::ByPackage(q))) => (p == q).then_some(self),
+            (F::ByModule(M::ByPackage(p)), F::ByFqName(q, _, _)) => (p == q).then_some(other),
```

Comment:
we select the more specific one given some module 0x2::coin and 0x2::coin::Coin<T>, whereas intersect implies that we'd select 0x2::coin. I do think it is nicer to opt for the more specific filter though, but then intersect may not be the best term to use here

Reply:
> we select the more specific one given some module 0x2::coin and 0x2::coin::Coin, whereas intersect implies that we'd select 0x2::coin.

Intersect implies that we would select the more specific query.  If I have the set of all types in the module `0x2::coin` and the set of all type instantiations of `0x2::coin::Coin` and I intersect them, I get the set of all type instantiations of `0x2::coin::Coin`.

Union would imply selecting `0x2::coin`.